backends/cuda/ceed-cuda-qfunction.c

```diff
@@ -183,17 +123,14 @@ int CeedQFunctionCreate_Cuda(CeedQFunction qf) {
   ierr = CeedQFunctionSetData(qf, data); CeedChkBackend(ierr);
 
   // Read source
-  char *source;
-  ierr = CeedQFunctionGetSourcePath(qf, &source); CeedChkBackend(ierr);
+  char *source_path;
+  ierr = CeedQFunctionGetSourcePath(qf, &source_path); CeedChkBackend(ierr);
   // Empty source path indicates user must supply Q-Function
-  if (source[0] != '\0') {
-    const char *funname = strrchr(source, ':') + 1;
-    data->qFunctionName = (char *)funname;
-    const int filenamelen = funname - source;
-    char filename[filenamelen];
-    memcpy(filename, source, filenamelen - 1);
-    filename[filenamelen - 1] = '\0';
-    ierr = CeedCudaLoadQFunction(qf, filename); CeedChkBackend(ierr);
+  if (source_path[0] != '\0') {
```

Comment:
I would prefer self-documented code, I would suggest making a function with a name that convey what is in the doc and that does that.